extensions-contrib/opencensus-extensions/src/main/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputRowParser.java

```diff
@@ -88,22 +107,30 @@ public OpenCensusProtobufInputRowParser withParseSpec(ParseSpec parseSpec)
       throw new ParseException(e, "Protobuf message could not be parsed");
     }
 
+    // Process metric descriptor labels map keys.
+    List<String> descriptorLabels = metric.getMetricDescriptor().getLabelKeysList().stream()
+        .map(s -> this.metricLabelPrefix + s.getKey())
+        .collect(Collectors.toList());
+
+    // Process resource labels map.
+    Map<String, Object> resourceLabelsMap = new HashMap<>(metric.getResource().getLabelsMap().entrySet().stream()
+        .collect(Collectors.toMap(entry -> this.resourceLabelPrefix + entry.getKey(),
+            Map.Entry::getValue)));
+
     final List<String> dimensions;
 
     if (!this.dimensions.isEmpty()) {
       dimensions = this.dimensions;
     } else {
-      Set<String> recordDimensions = metric.getMetricDescriptor().getLabelKeysList().stream()
-          .map(s -> s.getKey())
-          .collect(Collectors.toSet());
+      Set<String> recordDimensions = new HashSet<>(descriptorLabels);
 
       // Add resource map key set to record dimensions.
-      recordDimensions.addAll(metric.getResource().getLabelsMap().keySet());
+      recordDimensions.addAll(resourceLabelsMap.keySet());
 
-      // NAME, VALUE dimensions will not be present in labelKeysList or Metric.Resource map as they
-      // are derived dimensions, which get populated while parsing data for timeSeries hence add
-      // them to recordDimensions.
-      recordDimensions.add(NAME);
+      // MetricDimension, VALUE dimensions will not be present in labelKeysList or Metric.Resource
+      // map as they are derived dimensions, which get populated while parsing data for timeSeries
+      // hence add them to recordDimensions.
+      recordDimensions.add(this.metricDimension);
```

Comment:
nit, the `this.` prefix is not necessary here

Reply:
I find `this.` increases readability i.e. distinguish easily between local vs global variables hence leaving it as it is.

Reply:
we can leave as is, but the convention in the Druid codebase we typically only use this pattern when there is ambiguity between local and member variables, e.g. it often happens in the constructor. It can make the code harder to read when the code often refers to member variables, in which case it becomes unnecessarily verbose.
This PR doesn't seem consistent about it either (see https://github.com/confluentinc/druid/pull/14/files#diff-13a54faa5722482b59f3f83c9fdfd4deR93-R95 which doesn't use it)

Reply:
Point Taken, updated the code.